core/src/muxing.rs

```diff
@@ -105,6 +97,22 @@ pub trait StreamMuxer {
     /// >           properly informing the remote, there is no difference between this and
     /// >           immediately dropping the muxer.
     fn poll_close(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>>;
+
+    /// Poll for an event of the underlying connection.
+    ///
+    /// In addition to returning an event, this function may be used to perform any kind of background
+    /// work that needs to happen for the muxer to do its work. Implementations can rely on this
+    /// function to be called regularly and unconditionally.
+    fn poll_event(
```

Comment:
Not a strong opinion, but what do you think about just calling it `poll`?
- It emphasizes that this method can / should be used to drive background work
- It's consistent with `Swarm::poll`, `Transport::poll`, etc. which all serve the same functionality (driving background work and returning events that may happen)

Reply:
With the additional functionality (i.e. _do generic work_) I find `poll` more intuitive than `poll_event` as well.

Reply:
I am not strongly attached to `poll_event`. I decided for it to have consistency among the functions in `StreamMuxer`. Currently all of them have a qualifier that expresses, what they are polling for. If it is just called `poll` that is lost.

Maybe the asymmetry is good because the functions shouldn't be treated the same way? Thoughts?

Reply:
Maybe the docs also need some rewriting. I think the fact that this function can be used to do background work is more a conclusion rather than a requirement. What needs to be documented well is that this function will be called conditionally, whereas all the others may not be.

Reply:
> Not a strong opinion, but what do you think about just calling it `poll`?
> 
>  * It emphasizes that this method can / should be used to drive background work

Not that I necessarily disagree but how does it emphasize that more than the current name?

>  * It's consistent with `Swarm::poll`, `Transport::poll`, etc. which all serve the same functionality (driving background work and returning events that may happen)

All other interfaces also don't have any other `poll` functions so I am not sure that is such a good comparison.

Reply:
> > Not a strong opinion, but what do you think about just calling it `poll`?
> > 
> > * It emphasizes that this method can / should be used to drive background work
> 
> Not that I necessarily disagree but how does it emphasize that more than the current name?
> 
> > * It's consistent with `Swarm::poll`, `Transport::poll`, etc. which all serve the same functionality (driving background work and returning events that may happen)
> 
> All other interfaces also don't have any other `poll` functions so I am not sure that is such a good comparison.

I read `poll_<xyz>` always as "poll for xyz", as in "poll this method if you are interested in xyz". So callers may call this method, but if they are not interested in xyz they don't have to.
`poll` on the other hand (at least how we use it in `Swarm`, `Transport`) is used to drive background work. Even if the user is not interested in the produced events, they still must call it continuously.

Reply:
Hmm, that does make sense. Let's go with `poll` then :)

Reply:
> I read `poll_<xyz>` always as "poll for xyz", as in "poll this method if you are interested in xyz". So callers may call this method, but if they are not interested in xyz they don't have to.
> `poll` on the other hand (at least how we use it in `Swarm`, `Transport`) is used to drive background work. Even if the user is not interested in the produced events, they still must call it continuously.

Same mindset for me. Well putten into words.